ironfish/src/network/utils/serializers.ts

```diff
@@ -19,26 +20,26 @@ const BLOCK_TRANSACTIONS_LENGTH_BYTES = 2
 
 export function writeBlockHeader(
   bw: bufio.StaticWriter | bufio.BufferWriter,
-  header: SerializedBlockHeader,
+  header: BlockHeader,
 ): bufio.StaticWriter | bufio.BufferWriter {
   bw.writeU32(header.sequence)
   bw.writeHash(header.previousBlockHash)
   bw.writeHash(header.noteCommitment.commitment)
   bw.writeU32(header.noteCommitment.size)
   bw.writeHash(header.nullifierCommitment.commitment)
   bw.writeU32(header.nullifierCommitment.size)
-  bw.writeBytes(BigIntUtils.toBytesLE(BigInt(header.target), 32))
+  bw.writeBytes(BigIntUtils.toBytesLE(BigInt(header.target.targetValue.toString()), 32))
   bw.writeBytes(BigIntUtils.toBytesLE(BigInt(header.randomness), 8))
-  bw.writeU64(header.timestamp)
+  bw.writeU64(header.timestamp.getTime())
 
   Assert.isTrue(BigInt(header.minersFee) <= 0)
   bw.writeBytes(BigIntUtils.toBytesLE(-BigInt(header.minersFee), 8))
 
-  bw.writeBytes(GraffitiSerdeInstance.deserialize(header.graffiti))
+  bw.writeBytes(header.graffiti)
   return bw
 }
 
-export function readBlockHeader(reader: bufio.BufferReader): SerializedBlockHeader {
+export function readBlockHeader(reader: bufio.BufferReader): BlockHeader {
   const sequence = reader.readU32()
   const previousBlockHash = reader.readHash('hex')
```

Comment:
You can get rid of `'hex'` here and it'll pull in the block hash as bytes, same for the nullifier commitment, then that avoids needing to use the SerdeInstances

Reply:
Changed